File: 1720.decode-xo-red-array.py
#
# @lc app=leetcode id=1720 lang=python3
#
# [1720] Decode XORed Array
#

# @lc code=start
class Solution:
    def decode(self, encoded: List[int], first: int) -> List[int]:
        result = [first]
        for x in encoded:
            result.append(result[-1] ^ x)
        return result
# @lc code=end

File: js/1720.decode-xo-red-array.js
/*
 * @lc app=leetcode id=1720 lang=javascript
 *
 * [1720] Decode XORed Array
 */

// @lc code=start
/**
 * @param {number[]} encoded
 * @param {number} first
 * @return {number[]}
 */
var decode = function(encoded, first) {
  let arr = [first]
  for (let i=0;i<encoded.length;i++){
    let a = encoded[i]
    let b = arr[arr.length-1]
    arr.push(a^b)
  }
  return arr
};
// @lc code=end

